sky/packages/sky/lib/src/widgets/basic.dart

```diff
@@ -243,6 +243,38 @@ class Center extends Align {
     : super(key: key, shrinkWrap: shrinkWrap, child: child);
 }
 
+class CustomShiftedBox extends OneChildRenderObjectWidget {
```

Comment:
> I wish this were called CustomLayout, like CustomPaint

Yeah, we can pick a better name.  I'm not sure we want to use the name `CustomLayout` because this widget handles only a subset of the possible things you might want custom layout for.  For example, it only handles one child.

> I'm not sure why I need to provide a getInnerConstraints callback. Why isn't getSize() enough?

Having `getInnerConstraints` lets you provide things like a max height for the child but then still let the child have its preferred height.

> We use the qualifiers "inner" and "intrinsic" here and elsewhere. I'm not sure exactly what they mean. I would assume that and intrinsic quality is one that doesn't depend on other widgets. So "inner"? We seem to have entered the realm of the spiritual :-).

We can rename `getInnerConstraints` to `getConstraintsForChild`, which is probably clearer.  The term "intrinsic" referred to properties of the child regardless of its environment.  For example, how wide would this child prefer to be if it wasn't constrained at all.